### PR TITLE
catch up with Scala 2.11.2 version (the most recent scala virtualized version)

### PR DESCRIPTION
**Pull request by timxzl**
bump scala version to 2.11.2
remove MacroParadise and change older plugins like continuations to their most recent alternatives.
catch up with Scala 2.11.2 style, fixing errors and reducing warnings

### CODE REVIEW

File: core/src/main/scala/scala/js/exp/TraitsExp.scala [hunk 1]
```diff
@@ -35,7 +35,7 @@ trait TraitsExp extends Traits with ProxyExp {
     }
 
     val implClazz = Class.forName(traitClazz.getName + "$class")
-    val parents = traitClazz.getInterfaces.filter(_ != implicitly[Manifest[scala.ScalaObject]].erasure)
+    val parents = traitClazz.getInterfaces.filter(_ != implicitly[Manifest[AnyRef]].runtimeClass)
```
Comment (julienrf) [inline core/src/main/scala/scala/js/exp/TraitsExp.scala]: Why was this change needed?

**Comment by julienrf** [comment]
Hi,

Thanks for this!

I just left one comment. Can you address it, please?

BTW, just out of curiousity, in which context are you using js-scala?

File: core/src/main/scala/scala/js/exp/TraitsExp.scala (hunk 1 above)
Comment (timxzl) [inline core/src/main/scala/scala/js/exp/TraitsExp.scala]: because Manifest.erasure has been deprecated by .runtimeClass

**Comment by timxzl** [comment]
Hi @julienrf ,

Thank you for the prompt reply!

I have added a line comment in reply to yours.

We are starting a project which generates both C++ and Javascript code, and we would like to try out js-scala as the code generator for Javascript. Seems like js-scala is a very neat tool that fits our need.

Thanks a lot for the great work!

File: core/src/main/scala/scala/js/exp/TraitsExp.scala (hunk 1 above)
Comment (julienrf) [inline core/src/main/scala/scala/js/exp/TraitsExp.scala]: I was talking about the change of `scala.Object` for `AnyRef`

Comment (timxzl) [inline core/src/main/scala/scala/js/exp/TraitsExp.scala]: oh, that one. Because [ScalaObject](http://www.scala-lang.org/api/2.10.4/index.html#scala.ScalaObject) has been removed, and the closest thing to it is AnyRef.

Comment (timxzl) [inline core/src/main/scala/scala/js/exp/TraitsExp.scala]: To my understanding, the code
```scala
val parents = traitClazz.getInterfaces.filter(_ != implicitly[Manifest[ScalaObject]].runtimeClass)
```
is to filter out the uninteresting `ScalaObject` root interface that is definitely inherited by all classes we are interested in. Since [`ScalaObject`](http://www.scala-lang.org/api/2.10.4/index.html#scala.ScalaObject) has been [removed from the language](https://groups.google.com/forum/#!topic/scala-user/lLRc_NOboLU), it seems that we no longer need this filter. Or do you think we still want to filter out `AnyRef` as an uninteresting root interface?

Comment (julienrf) [inline core/src/main/scala/scala/js/exp/TraitsExp.scala]: Do [this test](https://github.com/js-scala/js-scala/blob/master/core/src/test/scala/scala/js/TestProxy.scala) still pass with your change?

Comment (timxzl) [inline core/src/main/scala/scala/js/exp/TraitsExp.scala]: Yes, it still passes.
```
> testOnly *TestProxy
[info] TestProxy:
[info] - testProxy
[info] Run completed in 682 milliseconds.
[info] Total number of tests run: 1
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 1, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
```

Comment (julienrf) [inline core/src/main/scala/scala/js/exp/TraitsExp.scala]: Great, so I merge your PR. Thanks again!

Comment (timxzl) [inline core/src/main/scala/scala/js/exp/TraitsExp.scala:38]: you're welcome! Thanks again for such a great project!
